test/ReqFailureUnitTests.hs

```diff
@@ -0,0 +1,68 @@
+module ReqFailureUnitTests (reqFailureUnitTests) where
+
+import Prelude ()
+
+import Game.LambdaHack.Core.Prelude
+
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+import           Game.LambdaHack.Common.Item
+import           Game.LambdaHack.Common.ItemAspect
+import           Game.LambdaHack.Common.ReqFailure
+import           Game.LambdaHack.Common.Time
+import           Game.LambdaHack.Content.ItemKind
+import qualified Game.LambdaHack.Content.RuleKind as RK
+import           Game.LambdaHack.Core.Dice
+import qualified Game.LambdaHack.Definition.Ability as Ability
+import           Game.LambdaHack.Definition.Color
+import           Game.LambdaHack.Definition.Defs
+import           Game.LambdaHack.Definition.Flavour
+
+import qualified Content.RuleKind
+
+reqFailureUnitTests :: TestTree
+reqFailureUnitTests = testGroup "reqFailureUnitTests" $
+    let testItemKind = ItemKind
+            { isymbol  = 'x'
+            , iname    = "12345678901234567890123"
+            , ifreq    = [ (UNREPORTED_INVENTORY, 1) ]  
+            , iflavour = zipPlain [Green]
+            , icount   = 1 + 1 `d` 2 
+            , irarity  = [(1, 50), (10, 1)]
+            , iverbHit = "hit"
+            , iweight  = 300
+            , idamage  = 1 `d` 1
+            , iaspects = [ AddSkill Ability.SkHurtMelee $ -16 * 5
+                        , SetFlag Ability.Fragile
+                        , toVelocity 70 ] 
+            , ieffects = []
+            , idesc    = "A really cool test item."
+            , ikit     = []
+            }
+        testItemFull = ItemFull 
+            { itemBase = Item 
+                { jkind = IdentityObvious (toContentId 667)
+                , jfid = Nothing
+                , jflavour = head (zipPlain [Black])  -- is there a less indirect way to do that?
```

Comment:
No and this is a good thing. :) The abstraction supports the loose character of game content declarations --- primarily as constraints on autobalancing (to some extent) PCG, not tools to micromanage brittle fixed hand-crafted content that needs to be rewritten or rebalanced every time other things change.

If really needed, we could add a function just for internal tests, hidden under `EXPOSE_INTERNAL`, but it's better to test via normal API, if not too troublesome.